cli/internal/cmd/iamdestroy.go

```diff
@@ -0,0 +1,158 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/edgelesssys/constellation/v2/cli/internal/cloudcmd"
+	"github.com/edgelesssys/constellation/v2/internal/cloud/gcpshared"
+	"github.com/edgelesssys/constellation/v2/internal/constants"
+	"github.com/edgelesssys/constellation/v2/internal/file"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+func runIAMDestroy(cmd *cobra.Command, _args []string) error {
+	log, err := newCLILogger(cmd)
+	if err != nil {
+		return fmt.Errorf("creating logger: %w", err)
+	}
+	defer log.Sync()
+	spinner := newSpinner(cmd.ErrOrStderr())
+	destroyer := cloudcmd.NewIAMDestroyer(cmd.Context())
+	fsHandler := file.NewHandler(afero.NewOsFs())
+
+	c := &destroyCmd{log: log}
+
+	return c.iamDestroy(cmd, spinner, destroyer, fsHandler)
+}
+
+type destroyCmd struct {
+	log debugLog
+}
+
+func (c *destroyCmd) iamDestroy(cmd *cobra.Command, spinner spinnerInterf, destroyer iamDestroyer, fsHandler file.Handler) error {
+	// check if there is a possibility that the cluster is still running by looking out for specific files
+	c.log.Debugf("Checking if %s exists", constants.AdminConfFilename)
+	_, err := fsHandler.Stat(constants.AdminConfFilename)
+	if !errors.Is(err, os.ErrNotExist) {
+		c.log.Debugf("File %s exists, cluster is probably running", constants.AdminConfFilename)
+		cmd.Printf("The file %s still exists, please make sure to terminate your cluster before destroying your IAM configuration.\n", constants.AdminConfFilename)
+		return nil
+	}
+	c.log.Debugf("Checking if %s exists", constants.ClusterIDsFileName)
+	_, err = fsHandler.Stat(constants.ClusterIDsFileName)
+	if !errors.Is(err, os.ErrNotExist) {
+		c.log.Debugf("File %s exists, cluster is probably running", constants.ClusterIDsFileName)
+		cmd.Printf("The file %s still exists, please make sure to terminate your cluster before destroying your IAM configuration.\n", constants.ClusterIDsFileName)
+		return nil
+	}
+
+	yes, err := cmd.Flags().GetBool("yes")
+	if err != nil {
+		return err
+	}
+
+	c.log.Debugf("\"yes\" flag is set to %t", yes)
+
+	gcpFileExists := false
+
+	c.log.Debugf("Checking if %s exists", constants.GCPServiceAccountKeyFile)
+	_, err = fsHandler.Stat(constants.GCPServiceAccountKeyFile)
+	if err != nil {
+		if !errors.Is(err, os.ErrNotExist) {
+			return err
+		}
+	} else {
+		c.log.Debugf("%s exists", constants.GCPServiceAccountKeyFile)
+		gcpFileExists = true
+	}
```

Comment:
```suggestion
	if errors.Is(err, os.ErrNotExist) {
		c.log.Debugf("%s exists", constants.GCPServiceAccountKeyFile)
		gcpFileExists = true
	} else if err != nil {
		return err
	} 
```

Reply:
If I'm not mistaken your proposal wouldn't work since it explicitly sets `gcpFileExists` to `true` if the file doesn't exist. I believe even with a logical not in front of `errors.Is(...)` this code wouldn't quite work out without having other if cases before it and then we are really similar to the code I wrote again. What do you think?